### PR TITLE
add a decorator for retrying zookeeper connections

### PR DESCRIPTION
**Pull request by Rob-Johnson**
we *should* be more resillient to single ZK boxes being
unavailable after this. PAASTA-6639

### CODE REVIEW

**Comment by Rob-Johnson** [comment]
@ronin13 this is what I came up with

File: paasta_tools/utils.py [hunk 1]
```diff
@@ -1700,3 +1701,45 @@ def prompt_pick_one(sequence, choosing):
         sys.exit(1)
     else:
         return result
+
+
+def retry(ExceptionToCheck, tries=4, delay=3, backoff=2, logger=None):
```
Comment (ronin13) [inline paasta_tools/utils.py]: As discussed, https://github.com/invl/retry / https://pypi.python.org/pypi/retry/ can be reused here.

File: paasta_tools/paasta_metastatus.py [hunk 2]
```diff
@@ -39,6 +40,11 @@
 from paasta_tools.utils import PaastaColors
 from paasta_tools.utils import print_with_indent
 
+
+logging.basicConfig()
+# kazoo can be really noisy - turn it down
+logging.getLogger("kazoo").setLevel(logging.CRITICAL)
```
Comment (mattmb) [inline paasta_tools/paasta_metastatus.py]: Was this info before? I agree it's annoyingly chatty, maybe ERROR would be okay though.

File: paasta_tools/mesos/master.py [hunk 3]
```diff
@@ -76,12 +81,13 @@ def fetch(self, url, **kwargs):
...
         return self.resolve(open(cfg[6:], "r+").read().strip())
 
+    @retry(KazooTimeoutError, tries=5, delay=0.5, logger=logger)
     def _zookeeper_resolver(self, cfg):
         hosts, path = cfg[5:].split("/", 1)
         path = "/" + path
 
         retry = KazooRetry(max_tries=10)
-        with zookeeper.client(hosts=hosts, read_only=True, connection_retry=retry) as zk:
+        with zookeeper.client(hosts=hosts, read_only=True, connection_retry=retry, command_retry=retry) as zk:
```
Comment (mattmb) [inline paasta_tools/mesos/master.py]: I don't really understand what these args are for. Isn't the decorator enough on its own?

Comment (Rob-Johnson) [inline paasta_tools/mesos/master.py]: nope - the decorator deals with a non-zookeeper exeception(ConnectionTimeout) - connection_retry and command_retry deal with zookeeper specific errors.

**Pull request merged**